test/ConfigManagerTest.cpp

```diff
@@ -0,0 +1,347 @@
+// Copyright 2023, University of Freiburg,
+// Chair of Algorithms and Data Structures.
+// Author: Andre Schlegel (April of 2023,
+// schlegea@informatik.uni-freiburg.de)
+
+#include <gtest/gtest.h>
+
+#include <cstddef>
+#include <vector>
+
+#include "util/ConfigManager/ConfigExceptions.h"
+#include "util/ConfigManager/ConfigManager.h"
+#include "util/ConfigManager/ConfigOption.h"
+#include "util/ConfigManager/ConfigShorthandVisitor.h"
+
+namespace ad_utility {
+TEST(ConfigManagerTest, GetConfigurationOptionByNestedKeysTest) {
+  ad_utility::ConfigManager config{};
+
+  // 'Compare' two configuration options
+  auto compareConfigurationOptions = []<typename T>(
+                                         const ad_utility::ConfigOption& a,
+                                         const ad_utility::ConfigOption& b) {
+    ASSERT_EQ(a.hasSetDereferencedVariablePointer(),
+              b.hasSetDereferencedVariablePointer());
+
+    if (a.hasSetDereferencedVariablePointer()) {
+      ASSERT_EQ(a.getValue<T>(), b.getValue<T>());
+      ASSERT_EQ(a.getValueAsString(), b.getValueAsString());
+    }
+  };
+
+  // Configuration options for testing.
+  int notUsed;
+
+  config.createConfigOption(
+      {"Shared_part", "Unique_part_1", "Sense_of_existence"}, "", &notUsed,
+      std::optional{42});
+  const ad_utility::ConfigOption& withDefault{
+      ad_utility::ConfigOption("Sense_of_existence", "", &notUsed, 42)};
```

Comment:
```suggestion
  const ad_utility::ConfigOption("Sense_of_existence", "", &notUsed, 42)};
```

(Similarly below)

Reply:
Took a stab at in [907f005](https://github.com/ad-freiburg/qlever/pull/977/commits/907f005ac72386c9efe7c340b2f9f6a175fb033f).

Reply:
Yes. I only think you can take by value here, I known that the `const&` implicitly extends the lifetime of the temporary, but still, why use this here. Of course it needs a name.
Yes, what you did there is exactly what I meant.